src/index.js

```diff
@@ -1,36 +1,118 @@
 import './style.css';
 
-const tasks = [
-  {
-    description: 'Wake up',
-    completed: true,
-    index: 0,
-  },
-  {
-    description: 'Pray',
-    completed: true,
-    index: 1,
-  },
-  {
-    description: 'Eat breakfast',
-    completed: false,
-    index: 2,
-  },
-];
-
+let tasks = [];
 const taskWrapper = document.querySelector('.to-dos');
+const newTask = document.querySelector('.new-task');
+const addNewTask = document.querySelector('.submit');
+const clearAll = document.querySelector('.clear-all');
 
-const showList = () => {
-  taskWrapper.innerHTML = '';
+const addToLocalStorage = () => {
+  localStorage.setItem('myTasks', JSON.stringify(tasks));
+};
+
+const getFromLocalStorage = () => {
+  if (localStorage.getItem('myTasks')) {
+    tasks = JSON.parse(localStorage.getItem('myTasks'));
+  }
+  return tasks;
+};
+
+const resetIndex = (tasks) => {
   for (let i = 0; i < tasks.length; i += 1) {
-    taskWrapper.innerHTML += `
-    <ul id="form">
-      <input type="checkbox" id="${tasks[i].index}" name="task" value="task">
-      <li for="${tasks[i].index}">${tasks[i].description}</li>
-      <i class="fa fa fa-times"></i><br>
-    </ul>
-    `;
+    let indx = i + 1;
+    indx = i;
+    tasks[i].index = indx;
+  }
+};
+
+const editTask = (desc, index) => {
+  tasks[index].description = desc;
+  addToLocalStorage();
+};
+
+const setState = (tasks, checkbox, index) => {
+  if (checkbox.checked === true) {
+    tasks[index].checked = true;
+  } else {
+    tasks[index].checked = false;
   }
 };
 
-window.addEventListener('load', showList);
\ No newline at end of file
+const displayTasks = () => {
+  taskWrapper.innerHTML = '';
+  const mylocal = getFromLocalStorage();
+  mylocal.forEach((tsk) => {
+    const li = document.createElement('li');
+    const checkbox = document.createElement('input');
+    checkbox.setAttribute('type', 'checkbox');
+    if (tsk.checked === true) {
+      checkbox.setAttribute('checked', 'checked');
+    }
+    checkbox.addEventListener('change', (e) => {
+      e.preventDefault();
+      setState(tasks, e.target, tsk.index);
+      addToLocalStorage();
+    });
```

Comment:
- [ ] I am able to see tasks that I added to the local storage but they are not displayed on the screen please check how you are creating your DOM elements
![dom](https://user-images.githubusercontent.com/44745855/167882121-3068f4c4-c4e5-43d7-9c09-0cc132ac8cab.png)

Reply:
- [ ] The property `index` should be set to the value of the new array length let us say if a 5th task is added to the list, the index of that task should equal to 5, check how your indexes are
![storage](https://user-images.githubusercontent.com/44745855/167883105-b7e3c710-47e2-43b4-be3e-f09988952cbc.png)